pkg/objstore/s3/s3.go

```diff
@@ -60,7 +68,16 @@ type Bucket struct {
 
 // NewBucket returns a new Bucket using the provided s3 config values.
 func NewBucket(logger log.Logger, conf []byte, reg prometheus.Registerer, component string) (*Bucket, error) {
-	var config Config
+	defIdleConnTimeout, _ := model.ParseDuration("90s")
```

Comment:
The idea here was to use `model.ParseDuration` so that we would less associate with the actual underlying type of `model.Duration` but I guess I will change it to just use `90 * time.Second` if that is OK with you. I agree that it definitely makes things clearer. As for other points, I agree. I thought it was the default and only acceptable behaviour of unmarshal function but we can move it into a function - it would be much nicer.

Reply:
:+1: